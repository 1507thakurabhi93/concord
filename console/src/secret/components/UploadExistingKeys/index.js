import React from 'react';
import { Field, reduxForm } from 'redux-form';
import FileInput from "../FileInput";

import { Form, Segment } from 'semantic-ui-react';

const UploadExistingKeys = props => {
  const { handleSubmit, pristine, submitting } = props;
  return (
    <Segment> 
      <h2>Upload Existing Keys</h2>
      <Form onSubmit={handleSubmit}>
        
        <Form.Field required>
          <label>Concord ID</label>
            <Field
              name="ConcordId"
              component={Form.Input}
              type="text"
              placeholder="Concord ID"
            />
        </Form.Field>

        {/* 
            Styling File Inputs
            http://jsfiddle.net/Dr_Dev/2nu1ngk5/ 
        */}

        <Form.Field required>
          <label>Public Key: </label>
            <Field
              name="publicKey"
              component={ FileInput }
            />
        </Form.Field>

        <Form.Field required>
          <label>Private Key: </label>
            <Field
              name="privateKey"
              component={ FileInput }
            />
        </Form.Field>

        <br></br>
        <Form.Button primary type="submit" disabled={pristine || submitting}>Submit</Form.Button>

      </Form>
    </Segment>
  );
};

export default reduxForm({
  form: 'UploadExistingKeys',
})(UploadExistingKeys);
