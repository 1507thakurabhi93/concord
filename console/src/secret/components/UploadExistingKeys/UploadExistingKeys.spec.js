import React from 'react';
import { shallow, mount } from 'enzyme';
import UploadExistingKeys from './';

import { Form } from 'semantic-ui-react';

describe('UploadExistingKeys Component', () => {

    it('should render with no errors', () => {
        shallow(<UploadExistingKeys />);
    });

    it('should render a text input for the Secret Name', () => {
        const wrapper = shallow(<UploadExistingKeys />);
        expect( wrapper ).toMatchSnapshot();
    });
    
});