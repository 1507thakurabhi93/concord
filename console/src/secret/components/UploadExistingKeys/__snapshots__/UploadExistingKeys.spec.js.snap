// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadExistingKeys Component should render a text input for the Secret Name 1`] = `
<Connect(Form(UploadExistingKeys))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="UploadExistingKeys"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldValidate={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;
